### ISSUE TITLE
False "copy" by track_copy(x) after 'x' is removed

### ISSUE THREAD

**Issue by HenrikBengtsson** [hadley/pryr#34]
The function returned by `track_copy()` falsely believe the object has been copied when it has actually been removed, e.g.
```r
> library('pryr')
> x <- 1; x[1] <- 1
> tx <- track_copy(x)
> x[2] <- 2
> tx()
x copied
> x[1] <- 2
> tx()
> rm(x)
> tx()
x copied
```
This is because it only compares the "address" before and after, i.e.
```r
> tx
function ()
{
    new <- address2(var, parent.frame())
    if (old == new)
        return(invisible(FALSE))
    if (!quiet)
        message(var, " copied")
    old <<- new
    invisible(TRUE)
}
```
Adding a
```r
if (!exists(var, envir=parent.frame(), inherits=FALSE))
  return(invisible(FALSE))
```
at the top would fix this.

**Comment by hadley** [comment]
Good catch - thanks!

**Comment by hadley** [comment]
Fixed in 94734ef66ed31238c37cdd47afe43f073d81bb0a
